packages/@aws-cdk/aws-route53-patterns/lib/website-redirect.ts

```diff
@@ -0,0 +1,73 @@
+import { DnsValidatedCertificate, ICertificate } from '@aws-cdk/aws-certificatemanager';
+import { CloudFrontWebDistribution, OriginProtocolPolicy, PriceClass, ViewerProtocolPolicy } from '@aws-cdk/aws-cloudfront';
+import { ARecord, IHostedZone, RecordTarget } from '@aws-cdk/aws-route53';
+import { CloudFrontTarget } from '@aws-cdk/aws-route53-targets';
+import { Bucket, RedirectProtocol } from '@aws-cdk/aws-s3';
+import { Construct, Fn, RemovalPolicy } from '@aws-cdk/core';
+
+export interface HTTPSRedirectProps {
+  /**
+   * HostedZone of the domain
+   */
+  readonly zone: IHostedZone;
+  /**
+   * The redirect target
+   */
+  readonly redirectTarget: string;
+  /**
+   * The domain name
+   *
+   * @default - the domain name of the zone
+   */
+  readonly domainName?: string;
+  /**
+   * The ARN of the certificate; Has to be in us-east-1
+   *
+   * @default - create a new certificate in us-east-1
+   */
+  readonly certificate?: ICertificate;
+}
+
+export class HTTPSRedirect extends Construct {
+  constructor(scope: Construct, id: string, props: HTTPSRedirectProps) {
+    super(scope, id);
+
+    const domainName = props.domainName || props.zone.zoneName;
+
+    const redirectCertArn = props.certificate ? props.certificate.certificateArn : new DnsValidatedCertificate(this, 'RedirectCertificate', {
+      domainName,
+      hostedZone: props.zone,
+      region: 'us-east-1',
```

Comment:
Will this work if the stack is not in us-east-1? If not, we should error.

Reply:
It works even if the stack is not deployed in `us-east-1`

https://github.com/aws/aws-cdk/blob/cbf41725bb13c5f75d1385cb70945fa8d7e3ce9c/packages/%40aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts#L18-L25

Reply:
Nice. good to know. thanks